tests/epyccel/test_arrays.py

```diff
@@ -3975,6 +3975,808 @@ def test_array_ndmin_2_order(language):
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int64, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_int64(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_float32, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_float32(bl)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_bool_to_other_types_3(language):
+    size = (2, 2)
+    bl = randint(0, 2, size = size, dtype= bool)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_float64, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_float64(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_cfloat, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_cfloat(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_cdouble, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_cdouble(bl)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_bool_to_other_types_4(language):
+    size = (2, 2)
+    bl = randint(0, 2, size = size, dtype= bool)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_pyint, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_pyint(bl)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_pyfloat, language=language)
+    assert epyccel_func(bl) == arrays.dtype_convert_to_pyfloat(bl)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_int8_to_other_types_1(language):
+    size = (2, 2)
+    integer8 = randint(low = iinfo('int8').min, high = iinfo('int8').max , size = size, dtype=np.int8)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int8, language=language)
+    assert epyccel_func(integer8) == arrays.dtype_convert_to_int8(integer8)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_bool, language=language)
+    assert epyccel_func(integer8) == arrays.dtype_convert_to_bool(integer8)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int16, language=language)
```

Comment:
These functions seem to be translated many times over these tests. Running the compiler is quite slow. It would be better to organise the tests in a way that means that each function only needs to be translated and compiled once.